libs/sam-formly-sample/src/lib/feature/formly-sample/formly-input.component.ts

```diff
@@ -1,144 +1,162 @@
-          label: 'Formly multi Select checkbox',
-          options: [
-            {
-              key: 'sports',
-              value: 'Sports'
-            },
-            {
-              key: 'movies',
-              value: 'Movies'
-            },
-            {
-              key: 'others',
-              value: 'Others'
-            }
-          ]
+     
+
+        {
+          key: 'level2',
+          type: 'autocomplete',
+          templateOptions: {
+            label: 'Auto Complete with multiselect',
+            service: this.service,
+            configuration: this.settings2,
+            model: this.autocompleteModel2,
+            modelChange: this.changes
+          },
         },
-      }
-        ],
-    },
+
+        {
+          key: 'level3',
+          type: 'autocomplete',
+          templateOptions: {
+            label: 'Auto Complete with disable with pre populated value',
+            service: this.service,
+            configuration: this.settings3,
+            model: this.autocompleteModel3,
+            modelChange: this.changes,
+            disabled: true
+
+           
+          },
+        }
+      ]
+    }
   ];
 
+
+  constructor(public service: AutocompleteSampleDataService) {
+    this.setup();
+  }
+
+  changes(value) {
+    console.log(value);
+  }
+
+  setup() {
+    this.settings.id = 'autocomplete1';
+    this.settings.primaryKeyField = 'id';
+    this.settings.primaryTextField = 'name';
+    this.settings.secondaryTextField = 'subtext';
+    this.settings.labelText = 'Autocomplete 1';
+    this.settings.selectionMode = SelectionMode.MULTIPLE;
+    this.settings.autocompletePlaceHolderText = 'Enter text';
+    this.settings.debounceTime = 0;
+
+
+    this.settings1.id = 'autocomplete1';
+    this.settings1.primaryKeyField = 'id';
+    this.settings1.primaryTextField = 'name';
+    this.settings1.secondaryTextField = 'subtext';
+    this.settings1.labelText = 'Autocomplete 1';
+    this.settings1.selectionMode = SelectionMode.SINGLE;
+    this.settings1.autocompletePlaceHolderText = 'Enter text';
+    this.settings1.debounceTime = 0;
+
+    this.settings2.id = 'autocomplete1';
+    this.settings2.primaryKeyField = 'id';
+    this.settings2.primaryTextField = 'name';
+    this.settings2.secondaryTextField = 'subtext';
+    this.settings2.labelText = 'Autocomplete 1';
+    this.settings2.selectionMode = SelectionMode.MULTIPLE;
+    this.settings2.autocompletePlaceHolderText = 'Enter text';
+    this.settings2.debounceTime = 0;
+
+
+    this.settings3.id = 'autocomplete1';
+    this.settings3.primaryKeyField = 'id';
+    this.settings3.primaryTextField = 'name';
+    this.settings3.secondaryTextField = 'subtext';
+    this.settings3.labelText = 'Autocomplete 1';
+    this.settings3.selectionMode = SelectionMode.MULTIPLE;
+    this.settings3.autocompletePlaceHolderText = 'Enter text';
+    this.settings3.debounceTime = 0;
+
+
+    this.autocompleteModel3.items.push(this.data[0]);

```

Comment:
Accessing arrays like this is generally unsafe. Prefer to use lodash/underscore .head method.

    function head(array) {
      return (array && array.length) ? array[0] : undefined;
    }